src/phoenix/experimental/evals/models/openai.py

```diff
@@ -254,6 +257,7 @@ def verbose_generation_info(self) -> str:
         return f"OpenAI invocation parameters: {self.public_invocation_params}"
 
     async def _async_generate(self, prompt: str, **kwargs: Any) -> str:
+        self.reload_client()
         invoke_params = self.invocation_params
         messages = self._build_messages(prompt, kwargs.get("instruction"))
         if functions := kwargs.get("functions"):
```

Comment:
Do we have a sense of how expensive this will be to re-initialize the client for each call?

Reply:
I haven't measured it, I'm going to spend some time to see if there's a nice way to reduce this overhead but at least this will work to solve the issues we've been seeing

Reply:
@axiomofjoy looks like it takes about 40ms per iteration. This is quite a bit of overhead but it is pretty low compared to the typical LLM API call, so it's probably negligible in practice (something like 1%-2% per call?)